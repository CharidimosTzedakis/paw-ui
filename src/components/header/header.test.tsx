import { describe, it, vi, expect } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import AppHeader from "./header";

vi.mock("wouter", async () => {
  const actual = await vi.importActual("wouter");

  return {
    ...actual,
    useLocation: vi.fn(() => ["/current-path", vi.fn()]),
  };
});

describe("<AppHeader/>", () => {
  it("should render correctly", async () => {
    const { container } = render(
      <AppHeader theme="light" toggleTheme={() => {}} />,
    );

    await waitFor(() => screen.getByText(/Cats/i));

    expect(container).toMatchSnapshot();
  });

  it("should toggle theme when theme switch is clicked", async () => {
    const toggleTheme = vi.fn();
    render(<AppHeader theme="light" toggleTheme={toggleTheme} />);

    await waitFor(() => screen.getByText(/Cats/i));

    const themeSwitch = screen.getByRole("switch", { checked: false });
    await userEvent.click(themeSwitch);

    expect(toggleTheme).toHaveBeenCalledTimes(1);
    expect(toggleTheme).toHaveBeenCalledWith(true, expect.anything());
  });
});
