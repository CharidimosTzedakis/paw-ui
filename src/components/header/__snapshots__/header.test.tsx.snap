// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppHeader/> > should render correctly 1`] = `
<div>
  <header
    class="ant-layout-header css-dev-only-do-not-override-1kf000u"
    style="display: flex; align-items: center;"
  >
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark css-dev-only-do-not-override-1kf000u"
      data-menu-list="true"
      role="menu"
      style="flex: 1; min-width: 0; font-size: 1.2rem; font-weight: 600;"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-menu-id="rc-menu-uuid-test-1"
        path="/cats"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/cats"
          >
            Cats
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        path="/breeds"
        role="menuitem"
        style="opacity: 1; order: 1;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/breeds"
          >
            Breeds
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        path="/favourites"
        role="menuitem"
        style="opacity: 1; order: 2;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/favourites"
          >
            Favourites
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
    <button
      aria-checked="false"
      class="ant-switch css-dev-only-do-not-override-1kf000u"
      role="switch"
      type="button"
    >
      <div
        class="ant-switch-handle"
      />
      <span
        class="ant-switch-inner"
      >
        <span
          class="ant-switch-inner-checked"
        >
          <span
            aria-label="moon"
            class="anticon anticon-moon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="moon"
              fill="currentColor"
              fill-rule="evenodd"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M489.5 111.66c30.65-1.8 45.98 36.44 22.58 56.33A243.35 243.35 0 00426 354c0 134.76 109.24 244 244 244 72.58 0 139.9-31.83 186.01-86.08 19.87-23.38 58.07-8.1 56.34 22.53C900.4 745.82 725.15 912 512.5 912 291.31 912 112 732.69 112 511.5c0-211.39 164.29-386.02 374.2-399.65l.2-.01zm-81.15 79.75l-4.11 1.36C271.1 237.94 176 364.09 176 511.5 176 697.34 326.66 848 512.5 848c148.28 0 274.94-96.2 319.45-230.41l.63-1.93-.11.07a307.06 307.06 0 01-159.73 46.26L670 662c-170.1 0-308-137.9-308-308 0-58.6 16.48-114.54 46.27-162.47z"
              />
            </svg>
          </span>
        </span>
        <span
          class="ant-switch-inner-unchecked"
        >
          <span
            aria-label="bulb"
            class="anticon anticon-bulb"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="bulb"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M632 888H392c-4.4 0-8 3.6-8 8v32c0 17.7 14.3 32 32 32h192c17.7 0 32-14.3 32-32v-32c0-4.4-3.6-8-8-8zM512 64c-181.1 0-328 146.9-328 328 0 121.4 66 227.4 164 284.1V792c0 17.7 14.3 32 32 32h264c17.7 0 32-14.3 32-32V676.1c98-56.7 164-162.7 164-284.1 0-181.1-146.9-328-328-328zm127.9 549.8L604 634.6V752H420V634.6l-35.9-20.8C305.4 568.3 256 484.5 256 392c0-141.4 114.6-256 256-256s256 114.6 256 256c0 92.5-49.4 176.3-128.1 221.8z"
              />
            </svg>
          </span>
        </span>
      </span>
    </button>
  </header>
</div>
`;
